src/helpers/ens.ts

```diff
@@ -33,25 +71,195 @@ const getBaseRegistrarImplementationContract = () => {
+  if (
+    Boolean(ensAssociatedRecord) &&
+    typeof ensAssociatedRecord === 'string' &&
+    parseInt(ensAssociatedRecord, 16) !== 0
+  ) {
+    data.push(
+      resolver.encodeFunctionData('setAddr(bytes32,address)', [
+        namehash,
+        ensAssociatedRecord,
+      ])
+    );
+  }
+  // content hash address
+  const contentHashAssociatedRecord = records.contentHash;
+  if (
+    Boolean(contentHashAssociatedRecord) &&
+    typeof contentHashAssociatedRecord === 'string' &&
+    parseInt(contentHashAssociatedRecord, 16) !== 0
+  ) {
+    data.push(
+      resolver.encodeFunctionData('setContenthash', [
+        namehash,
+        contentHashAssociatedRecord,
+      ])
+    );
+  }
+  // coin addresses
+  const coinAddressesAssociatedRecord = records.coinAddress;
+  if (coinAddressesAssociatedRecord) {
+    data.push(
+      coinAddressesAssociatedRecord.map(coinRecord => {
+        const { decoder, coinType } = formatsByName[coinRecord.key];
+        let addressAsBytes;
+        if (!coinRecord.address || coinRecord.address === '') {
+          addressAsBytes = Buffer.from('');
+        } else {
+          addressAsBytes = decoder(coinRecord.address);
+        }
+        return resolver.encodeFunctionData('setAddr(bytes32,uint256,bytes)', [
+          namehash,
+          coinType,
+          addressAsBytes,
+        ]);
+      })
+    );
+  }
+  // text addresses
+  const textAssociatedRecord = records.text;
+  if (textAssociatedRecord) {
+    data.push(
+      textAssociatedRecord.map(textRecord => {
+        return resolver.encodeFunctionData('setText', [
+          namehash,
+          textRecord.key,
+          textRecord.value,
+        ]);
+      })
+    );
+  }
+  // flatten textrecords and addresses and remove undefined
+  return data.flat().filter(Boolean);
+};
+
+const generateSalt = () => {
+  const random = new Uint8Array(32);
+  crypto.getRandomValues(random);
+  const salt =
+    '0x' +
+    Array.from(random)
+      .map(b => b.toString(16).padStart(2, '0'))
+      .join('');
+  return salt;
+};
+
+const getENSExecutionDetails = async ({
+  name,
+  type,
+  ownerAddress,
+  rentPrice,
+  duration,
+  records,
+}: {
+  name: string;
+  type: ENSRegistrationTransactionType;
+  ownerAddress?: string;
+  rentPrice?: string;
+  duration?: number;
+  records?: ENSRegistrationRecords;
+}): Promise<{
+  methodArguments: (string | string[] | number)[] | null;
+  value: BigNumberish | null;
+  contract: Contract | null;
+}> => {
+  let args: (string | string[] | number)[] | null = null;
+  let value: string | null = null;
+  let contract: Contract | null = null;
+  switch (type) {
+    case ENSRegistrationTransactionType.COMMIT: {
+      if (!name || !ownerAddress) throw new Error('Bad arguments for commit');
+      const salt = generateSalt();
+      const registrarController = getENSRegistrarControllerContract();
+      const commitment = await registrarController.makeCommitment(
```

Comment:
Aren't we supposed to use `makeCommitmentWithConfig` with `registerWithConfig`?

Reply:
yep you're right, thanks for the catch